source/core/index.ts

```diff
@@ -877,11 +876,8 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 
 		try {
 			// Errors are emitted via the `error` event
-			const rawBody = await getStreamAsBuffer(from);
-
-			// TODO: Switch to this:
-			// let rawBody = await from.toArray();
-			// rawBody = Buffer.concat(rawBody);
+			const fromArray = await from.toArray();
+			const rawBody = fromArray.every(b => isBuffer(b)) ? Buffer.concat(fromArray) : Buffer.from(fromArray.join(''));
```

Comment:
Why are you checking every chunk for whether it's a buffer? I don't think it can be mixed.

Reply:
You're probably right, should be safe to just check the first item. Updated!